fuel-vm/src/interpreter/contract/tests.rs

```diff
@@ -53,19 +55,49 @@ fn test_contract_balance(b: Word, c: Word) -> Result<(), RuntimeError> {
 }
 
 #[test_case(true, 0, 50, 32 => Ok(()); "Can transfer from external balance")]
-fn test_transfer(external: bool, a: Word, b: Word, c: Word) -> Result<(), RuntimeError> {
+#[test_case(false, 0, 50, 32 => Ok(()); "Can transfer from internal balance")]
+#[test_case(true, 0, 70, 32 => Err(RuntimeError::Recoverable(PanicReason::NotEnoughBalance)); "Cannot transfer from external balance insufficient funds")]
+#[test_case(false, 0, 70, 32 => Err(RuntimeError::Recoverable(PanicReason::NotEnoughBalance)); "Cannot transfer from internal balance insufficient funds")]
+fn test_transfer(
+    external: bool,
+    contract_id_offset: Word,
+    transfer_amount: Word,
+    asset_id_offset: Word,
```

Comment:
It would be nice if you tested something with `contract_id_offset` and `asset_id_offset` arguments. Otherwise, it is strange why do we have them=D

Reply:
I added some simple tests that swap them. I think that's good enough because it shows that it actually uses memory. There are infinite other tests we could write including the offsets and none of them seem particularly interesting off the top of my head.

Reply:
It would be nice to test some bad cases like memory overflow=)